src/kernels/composable_kernel/include/tensor_operation/xdlops_gemm.hpp

```diff
@@ -663,30 +754,16 @@ struct XdlopsGemm_t
                 for(index_t k_i      = 0; k_i < K; ++k_i)
                     b[k_i + n_i * K] = p_b_wave[k_i * N + laneId + NPerXdlops * n_i];
 
-            // get pointer of registers
-            auto pa = reinterpret_cast<const data_type*>(&a);
-            auto pb = reinterpret_cast<const data_type*>(&b);
-
-            for(index_t m_i = 0; m_i < MRepeats; ++m_i)
-            {
-
-                for(index_t n_i = 0; n_i < NRepeats; ++n_i)
-                {
-
 #if CK_WORKAROUND_SWDEV_229564
 #pragma unroll
 #endif
-                    for(index_t k_i = 0; k_i < K; ++k_i)
-                    {
-                        for(index_t i = 0; i < nxdlops; ++i)
-                            mfma_type.template run<MPerXdlops, NPerXdlops>(
-                                &pa[(k_i * nxdlops + i) * mfma_type.k_base +
-                                    m_i * K * nxdlops * mfma_type.k_base],
-                                &pb[(k_i * nxdlops + i) * mfma_type.k_base +
-                                    n_i * K * nxdlops * mfma_type.k_base],
-                                p_c_thread + (NRepeats * m_i + n_i) * GetRegSizePerXdlops());
-                    }
-                }
+            for(index_t k_i = 0; k_i < K * KRepeats; ++k_i)
+            {
+                p_c_thread = mfma_type.template run<MPerXdlops * MRepeats,
+                                                    NPerXdlops * NRepeats,
+                                                    AStride,
+                                                    BStride>(
+                    &pa[k_i * mfma_type.k_base], &pb[k_i * mfma_type.k_base], p_c_thread);
```

Comment:
```pa``` still seems to be a pointer to scalar, not vector. Am I right?

Reply:
Since `a/b/pa/pb` are normally small, I don't expect defining them as vectors would make much differences.

Perhaps it can be a future PR.

Reply:
@whchung If compiler can optimize pointer access out entirely, maybe it's better to keep using pointer in C++?

Reply:
True. So I think it's prudent to keep the PR in its current fashion. And file a separate PR should it be proven moving `a/b/pa/pb` to vector types really helps.